Triton_Inference_Server_Python_API/examples/kafka-io/README.md

```diff
@@ -0,0 +1,145 @@
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+-->
+
+# Triton Inference Server Kafka I/O Deployment
+
+Using the Triton Inference Server In-Process Python API you can
+integrate triton server based models into any Python framework
+to consume the messages from a Kafka topic and produce the inference results
+back to the kafka topic of choice.
+
+This directory contains an example Triton Inference Server
+deployment based on Kafka I/O that uses threads for each of server, consumer and producer.
+
+| [Installation](#installation) | [Run Deployment](#starting-the-pipeline) | [Send Requests](#send-requests-to-deployment) |
+
+
+## Installation
+
+In this Kafka I/O pipeline we deploy a pre-processing stage of tokenization based on `transformers` tokenization module and can be extended to any type of models as needed.
+
+
+### Clone Repository
+```bash
+git clone https://github.com/triton-inference-server/tutorials.git
+cd tutorials/Triton_Inference_Server_Python_API/examples/kafka-io
+```
+
+### Install dependencies
+
+Please note the following command will take many minutes depending on
+your hardware configuration and network connection.
+
+
+```bash
+pip install -r requirements.txt
+```
+
+If triton server is not already installed, install the dependency by using the following command.
+
+```bash
+pip install /opt/tritonserver/python/tritonserver-2.44.0-py3-none-any.whl
+```
+
+In order to run kafka server, we need to download and start the kafka service, execute the below script that starts the kafka service in the background by doing the following:
+1. Download kafka
+2. Start Kafka service by starting Zookeeper and Kafka brokers
+3. Create 2 new topics with names `inference-input` as input queue and `inference-output` to store the inference results
+
+```bash
+chmod +x start-kafka.sh
+./start-kafka.sh
+```
+
+In case, you see any issues starting the server, make sure the system has java installed. If there is no java installed, please follow the instructions below to install java.
+
+```bash
+apt-get update
+apt-get upgrade
+apt install openjdk-17-jdk openjdk-17-jre
```

Comment:
I did encounter issues running the `start-kafka.sh` script due to missing these packages. Since our aim is to leverage a container to run this tutorial, I think we should add these packages to the `requirements.txt`, with their versions pinned.

Reply:
Taken care of this by moving the java installation into kafka script.